shared/README.md

```diff
@@ -0,0 +1,91 @@
+
+Instead by using an approach similar to vendoring, where each dist ships with a copy of the code of the version it know it works with it allows them to co-exist in the same env without problems, and also reduces the cognitive load on developers when making changes to the shared libs -- as long as it passes the CI it should be good (and we don't have to test an ever increasing matrix of versions to ensure compatibility.)
+
+### Shared libraries _must_ use relative imports for other shared libraries**
+
+The one caveat to this is is due to the side-effect that these shared modules are going to be imported from different locations (for example `airflow._shared.timezones.timezone` and `airflow.sdk._shared.timezones.timezone`) then any imports inside the shared code to other parts of shared libraries _must_ make use of relative imports, and possibly needs `# noqa: TID252`
+
+For example, to use the shared timezone library from another shared library, lets say `shared/logging/src/airflow_shared/logging/config.py` you would have
+
+```python
+from ..timezones.timezone import is_naive  # noqa: TID252
+```
+
+This is different to the rest of the airflow codebase where relative imports are not allowed.
+
+### Shared libraries should be grouped one level beneath `airflow_shared/`
+
+In order to make both the relative imports above function and to provider simple structure, we want to arrange things in single named chunks, where the folder name is also the "shared library name". For example
+
+
+```
+shared
+├── README.md
+└── timezones/
+    ├── pyproject.toml
+    ├── src/
+    │   └── airflow_shared
+    │       └── timezones/
+    │           ├── __init__.py
+    └── tests/
+└── logging/
+    ├── pyproject.toml
+    ├── src/
+    │   └── airflow_shared
+    │       └── logging/
+    │           ├── __init__.py
+    └── tests/
+```
+
+## Use In Editable installs/Git checkouts
+
+In the editable checkout (which is what you get by default when using `uv sync` etc), the shared libraries are symlinked directly in place, for example `task-sdk/src/airflow/sdk/_shared/timezones` is symlinked to `shared/timezones/src/airflow_shared/timezones`. This means that python automatically finds the files underneath the module as if they lived directly under `airflow.sdk._shared.timezones.X`.
+
+
+### Including In built distributions
+
+Symlinks will not work when we build the main distributions for PyPI and other channels (since the symlinks would be to files outside of the distribution otherwise).
```

Comment:
```suggestion
Symlinks will not work when we build the main distributions for PyPI and other channels (since the symlinks would be to files outside of the distribution otherwise). This is specific to sdist; building wheel resolves the symlinks.
```